storage-operator/pkg/salt/client.go

```diff
@@ -206,11 +207,32 @@ func (self *Client) PollJob(
 		return nil, fmt.Errorf("Salt job %s failed to run", jobId)
 	default:
 		jobLogger.Info("Salt job failed")
-		reason := nodeResult["return"].(string)
+		reason := getStateFailureRootCause(nodeResult["return"])
 		return nil, &AsyncJobFailed{reason}
 	}
 }
 
+func getStateFailureRootCause(output interface{}) string {
+	const non_root_error_prefix string = "One or more requisite failed"
+
+	switch error := output.(type) {
+	case string:
+		return error
+	case map[string]interface{}:
+		for state := range error {
```

Comment:
Oh boy that's confusing... Can we have `state, _` or does Go not like that?

Reply:
renamed the variable to be more explicit